core/chains/evm/client/rpc_client_test.go

```diff
@@ -184,6 +227,133 @@ func TestRPCClient_SubscribeNewHead(t *testing.T) {
 		require.ErrorContains(t, err, "RPCClient returned error (rpc)")
 		tests.AssertLogEventually(t, observed, "evmclient.Client#EthSubscribe RPC call failure")
 	})
+	t.Run("Closed rpc client should remove existing SubscribeNewHead subscription with WS", func(t *testing.T) {
```

Comment:
All `Closed rpc client should remove existing...` tests have a lot in common. Could you try minimize amount of code duplication?